### PR TITLE
Add libltc 1.3.1

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **libltc/1.3.1**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by MartinDelille** [comment]
@x42 FYI

File: recipes/libltc/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
```
Comment (ruilvo) [inline recipes/libltc/all/conanfile.py:46]: In the `argtable2` recipe (that I choose at random when searching for autotools in CCI), I found this:
```python
self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
```
Try adding that `win_bash=tools.os_info.is_windows` to the constructor too :)

**Review by ruilvo — CHANGES_REQUESTED** [review]

**Comment by ruilvo** [comment]
You need
```python
    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC
```

File: recipes/libltc/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:41]:
```suggestion
    tools.get(**self.conan_data["sources"][self.version],
              destination=self._source_subfolder, strip_root=True)
```

File: recipes/libltc/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        args = []
+        if self.options.shared:
+            args.extend(["--disable-static", "--enable-shared"])
+        else:
+            args.extend(["--disable-shared", "--enable-static"])
+        self._autotools.configure(args=args)
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:52]:
```suggestion
        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
```

So you don't need chdir

File: recipes/libltc/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.make()
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:58]:
```suggestion
        autotools = self._configure_autotools()
        autotools.make()
```

File: recipes/libltc/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.install()
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:64]:
```suggestion
        autotools = self._configure_autotools()
        autotools.install()
```

File: recipes/libltc/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.unlink(os.path.join(self.package_folder, "lib", "libltc.la"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["ltc"]
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:70]:
```suggestion
        self.cpp_info.libs = ["ltc"]
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

```
$ ldd /tmp/install/lib/libltc.so.11
	linux-vdso.so.1 (0x00007ffd19782000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f46bca4e000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f46bc882000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f46bcbc4000)
```

File: recipes/libltc/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.unlink(os.path.join(self.package_folder, "lib", "libltc.la"))
+
+    def package_info(self):
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:69]:
```suggestion
    def package_info(self):
        self.cpp_info.names["pkg_config"] = "ltc"
```

https://github.com/x42/libltc/blob/v1.3.1/ltc.pc.in

**Comment by ruilvo** [comment]
> libltc/1.3.1: WARN: Error running `configure --help`: Cannot recognize the Windows subsystem, install MSYS2/cygwin or specify a build_require to apply it

Yap, you'll need MSYS2 as a build dependency

File: recipes/libltc/all/conanfile.py [hunk 8]
```diff
@@ -35,6 +35,10 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
+    def build_requirements(self):
+        if tools.os_info.is_windows and self.settings.compiler != "Visual Studio":
```
Comment (ruilvo) [inline recipes/libltc/all/conanfile.py:39]:
```suggestion
        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```
You'll probably need it like this (I copied from other recipes).
And you'll need to set the environment to use the cl compiler,
From automake recipe:
```python
    @contextmanager
    def _build_context(self):
        if self.settings.compiler == "Visual Studio":
            with tools.vcvars(self.settings):
                with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",}):
                    yield
        else:
            yield
```

**Review by ruilvo — CHANGES_REQUESTED** [review]

**Comment by ruilvo** [comment]
> checking for gcc... gcc
> checking whether the C compiler works... no
Yap, you have some flags to set :)

**Comment by MartinDelille** [comment]
Windows debug build is still failing. Here is an extract from the logs:

```
libtool: compile:  /c/j/w/buildsinglereference@2/.conan/data/automake/1.16.2/_/_/package/3e48e69237f7f2196164383ef9dedf0f93cbf249/bin/share/automake-1.16/compile cl -nologo -DHAVE_CONFIG_H -I. -I/c/j/w/buildsinglereference@2/.conan/data/libltc/1.3.1/_/_/build/b2b69efcc79a149e0bc765fb12ba0278d52fc260/source_subfolder/src -Wall -Zi -Ob0 -Od -MTd -c -showIncludes /c/j/w/buildsinglereference@2/.conan/data/libltc/1.3.1/_/_/build/b2b69efcc79a149e0bc765fb12ba0278d52fc260/source_subfolder/src/decoder.c -o libltc_la-decoder.obj
decoder.c
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt.h(212): warning C4668: '__cplusplus' is not defined as a preprocessor macro, replacing with '0' for '#if/#elif'
```

**Comment by ruilvo** [comment]
At least it already started building, you're in the right track. Let's examine the logs.

File: recipes/libltc/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        args = []
+        if self.options.shared:
+            args.extend(["--disable-static", "--enable-shared"])
+        else:
+            args.extend(["--disable-shared", "--enable-static"])
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:72]:
```suggestion
            args.extend(["--disable-shared", "--enable-static"])
        if self.settings.compiler == "Visual Studio":
            self._autotools.cxx_flags.append("-EHsc")
            self._autotools.flags.append("-FS")
```

It fixes your problem with MTd

**Comment by uilianries** [comment]
Almost there, the libraries names are different when built by msvc:

> ibltc/1.3.1 package(): Packaged 1 '.dll' file: ltc-11.dll
libltc/1.3.1 package(): Packaged 1 '.h' file: ltc.h
libltc/1.3.1 package(): Packaged 1 '.lib' file: ltc.dll.lib

**Comment by MartinDelille** [comment]
Shall I rename them or change something in package_info()  ?

File: recipes/libltc/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.unlink(os.path.join(self.package_folder, "lib", "libltc.la"))
+
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "ltc"
+        self.cpp_info.libs = ["ltc"]
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:95]:
```suggestion
        self.cpp_info.libs = ["ltc.dll.lib"] if self.settings.compiler == "Visual Studio" and self.options.shared else ["ltc"]
```

**Comment by ruilvo** [comment]
@MartinDelille can I ask you please to write something in the commit message other than default? I'm receiving notifications for this PR and it's hard to know what happened :)

**Comment by MartinDelille** [comment]
@ruilvo Yes sorry I'll write more explicit message in the future! Sorry for the noise and thank you for your support with this recipe! 😄

File: recipes/libltc/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.unlink(os.path.join(self.package_folder, "lib", "libltc.la"))
+
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "ltc"
+        self.cpp_info.libs = ["ltc.dll.lib"] if self.settings.compiler == "Visual Studio" and self.options.shared else ["ltc"]
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:95]:
```suggestion
        self.cpp_info.libs = ["ltc"]
```

File: recipes/libltc/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            autotools.make()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.unlink(os.path.join(self.package_folder, "lib", "libltc.la"))
```
Comment (uilianries) [inline recipes/libltc/all/conanfile.py:91]:
```suggestion
        os.unlink(os.path.join(self.package_folder, "lib", "libltc.la"))
        if self.settings.compiler == "Visual Studio" and self.options.shared:
            tools.rename(os.path.join(self.package_folder, "lib", "ltc.dll.lib"),
                         os.path.join(self.package_folder, "lib", "ltc.lib")
```

**Review by uilianries — COMMENTED** [review]
Conan hooks is not happy with that name, let's use a regular library name

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
Finally working! :D

**Review by prince-chrismc — APPROVED** [review]

**Comment by MartinDelille** [comment]
Thank you again for your kind help on this recipe!

**Pull request merged**